lib/src/helpers/adaptive_controls.dart

```diff
@@ -1,13 +1,32 @@
 import 'package:chewie/chewie.dart';
 import 'package:flutter/material.dart';
 
+enum ControlsType { cupertino, material, materialDesktop, adaptive }
+
 class AdaptiveControls extends StatelessWidget {
   const AdaptiveControls({
     Key? key,
+    required this.controlsType,
   }) : super(key: key);
 
+  final ControlsType controlsType;
+
   @override
   Widget build(BuildContext context) {
+    switch (controlsType) {
+      case ControlsType.cupertino:
+        return const CupertinoControls(
+          backgroundColor: Color.fromRGBO(41, 41, 41, 0.7),
+          iconColor: Color.fromARGB(255, 200, 200, 200),
+        );
+      case ControlsType.material:
+        return const MaterialControls();
+      case ControlsType.materialDesktop:
+        return const MaterialDesktopControls();
+      case ControlsType.adaptive:
+        break;
+    }
+
```

Comment:
Instead of having to duplicate code in this method, consider doing the following:

* Instead of using a separate enum for this, you may want to use an optional and nullable `TargetPlatform` when creating this widget. That way if it exists, use that value, otherwise, fallback to the one from the theme.
* Then you can determine the effective `TargetPlatform` to use based on the above and pass that into the switch statement:

```dart
final effectiveTargetPlatform = customPlatform ?? Theme.of(context).platform;
switch (effectiveTargetPlatform) {
  //...
}
```

Reply:
@vinaykumarrock986612 any updates?